From e82305147b2622f5f9e5115c272e33159de4ca72 Mon Sep 17 00:00:00 2001
From: Max Filippov <jcmvbkbc@gmail.com>
Date: Sat, 20 Aug 2016 00:20:41 -0700
Subject: [PATCH 2/2] xtensa: report stack usage

2016-08-21  Max Filippov  <jcmvbkbc@gmail.com>
gcc/
	* config/xtensa/xtensa.c (xtensa_expand_prologue): Update
	current_function_static_stack_size variable with the static
	stack frame size of the current function when
	flag_stack_usage_info is enabled.
---
 gcc/config/xtensa/xtensa.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gcc/config/xtensa/xtensa.c b/gcc/config/xtensa/xtensa.c
index 527fe88..f4504bf 100644
--- a/gcc/config/xtensa/xtensa.c
+++ b/gcc/config/xtensa/xtensa.c
@@ -2679,6 +2679,9 @@ xtensa_expand_prologue (void)
 
   total_size = compute_frame_size (get_frame_size ());
 
+  if (flag_stack_usage_info)
+    current_function_static_stack_size = total_size;
+
   if (TARGET_WINDOWED_ABI)
     {
       if (total_size < (1 << (12+3)))
-- 
2.1.4

